src/pages/preferences/preferencesTabs/notificationTab/utils.tsx

```diff
@@ -0,0 +1,84 @@
+import styled from "@emotion/styled";
+import { LeafyGreenTableRow } from "@leafygreen-ui/table/new";
+import { getPatchRoute, getTaskRoute, getVersionRoute } from "constants/routes";
+import { size } from "constants/tokens";
+import { GeneralSubscription, Selector } from "gql/generated/types";
+import { ResourceType } from "types/triggers";
+
+export const processSubscriptionData = (
+  subscriptions: GeneralSubscription[],
+  globalSubscriptionIds: Set<string>
+) => {
+  if (!subscriptions || !subscriptions.length) {
+    return;
+  }
+
+  return (
+    subscriptions
+      // Filter out a user's global subscriptions for tasks, spawn hosts, etc.
+      .filter(({ id }) => !globalSubscriptionIds.has(id))
+      // For subscriptions that contain regex selectors or additional trigger data, append an expandable section
+      .map((subscription) => {
+        const hasTriggerData = !!Object.entries(subscription.triggerData ?? {})
+          .length;
+        const hasRegexSelectors = !!subscription.regexSelectors.length;
+        return hasTriggerData || hasRegexSelectors
+          ? {
+              ...subscription,
+              renderExpandedContent: (
+                row: LeafyGreenTableRow<GeneralSubscription>
+              ) => (
+                <ExpandedBlock data-cy="expanded-block">
+                  {hasTriggerData && (
+                    <div data-cy="trigger-data">
+                      {JSON.stringify(row.original.triggerData, null, 2)}
+                    </div>
+                  )}
+                  {hasRegexSelectors && (
+                    <div data-cy="regex-selectors">
+                      {JSON.stringify(
+                        formatRegexSelectors(row.original.regexSelectors),
+                        null,
+                        2
+                      )}
+                    </div>
+                  )}
+                </ExpandedBlock>
+              ),
+            }
+          : subscription;
+      })
+  );
+};
+
+const ExpandedBlock = styled.pre`
+  padding: ${size.s} ${size.l};
+`;
+
+export const getResourceRoute = (resourceType: ResourceType, id: string) => {
```

Comment:
I think this function might need to be updated to link to the project (linking to either the project patches page or project health page seems fine to me)

<img width="666" alt="Screenshot 2023-05-24 at 10 07 35 AM" src="https://github.com/evergreen-ci/spruce/assets/47064971/b895b9fb-4674-4eb4-88c4-386bf51f5ac4">


In case it's needed as a reference, I think this is the part that distinguishes project subscriptions: [[link]](https://github.com/evergreen-ci/spruce/blob/24907af333c612312bfb303255940bd0e95f65b0/src/components/Notifications/utils.ts#L80-L89)